# Genome Wide Association Analysis (GWAS)
---- 
#### How the app performs the GWAS analysis:

    This application utilizes a custom WebAssembly module specifically designed for in browser GWAS analysis. 
    WebAssembly is a web technology that allows compiling code written in languages like C++ to run efficiently within web browsers. 
    This enables complex computations like GWAS to be performed directly in the browser without requiring users to download and install software.
    This WebAssembly module leverages functionalities from PLINK 1.07, a widely used open-source software suite for whole-genome association studies. 
    PLINK provides tools for data management, quality control, and various GWAS analysis methods. For further details, please refer to [Plink documentation](https://zzz.bwh.harvard.edu/plink/)

#### GWAS analysis without correction for population structure:

    This approach utilizes the --assoc option in plink, likely performing a simple chi-square test for each SNP to assess its association with the chosen camelina phenotype (trait) data.
    No explicit correction for population stratification is applied. This method can be faster but is susceptible to identifying false positives due to ancestry differences.

#### GWAS analysis with correction for population structure:

    This approach employs a linear regression model (--linear) with covariate adjustments (--covar).
    Additional principal components (PCs) or eigenvectors from a separate analysis are included as covariates. These components capture genetic variations due to ancestry and are used to account for population structure in the association test.
    This method is more robust and helps to reduce spurious associations arising from population stratification.

#### Phenotype Data Selection:

    The application also offers flexibility by allowing users to select various camelina phenotype datasets for GWAS analysis. This enables researchers to investigate the genetic underpinnings of different traits in camelina.

#### Additional Notes:

    ---


#### How to perform GWAS in your browser:

   - Choose either of the with or without correction option
   - Press run


import { Steps } from 'nextra/components'
 
<Steps>

###  Navigate to [GWAS module](https://www.camelina-hub.org/router?component=gwas)
![Hello](/GWAS_HomeScreem.png)

### Select a trait
![Hello](/GWAS_2_select_a_trait.png)

### Choose one of the with or without correction option
![Hello](/GWAS_3_SelectCorrection.png)

### Click run
![Hello](/GWAS_4_ClickRun.png)


### Track the progress 
![Hello](/GWAS_5_progress.png)
![Hello](/GWAS_6_progress.png)


### Visualizations/Results 
    After the completion of GWAS analysis, the following plots are created based the selected phenotype.

    - [x] Manhattan Plot
    - [x] Qunatile quantile plot
    - [x] Functional gene Annotation
    - [x] Multiomics data analysis with genome browser

    Click [Results](/modules/GWAS/Results) to read more on GWAS results.

</Steps>